### ISSUE TITLE
[QUESTION] How does the performance hack called "Disable Framebuffer" work?

### ISSUE THREAD

**Issue by Adamillo** [gonetz/GLideN64#2799]
I've been curious to know how this even works, since most GPU's require a framebuffer to even be able to work and render things at all

**Comment by cruduxcru0** [comment]
Turns off render to texture emulation, which in the N64's case is manipulation of in-memory screen buffers. Many many games use this as the cornerstone of special effects.

Modern games still use render to texture extensively, even more so for anti aliasing like TAA, deferred render queues like in Doom Eternal etc.

**Comment by Adamillo** [comment]
> It helps now that modern APIs like Vulkan and DX12 have now finally closed that gap, with Vulkan having now direct RAM access with special API functions, so you can do things on CPU and GPU with zero performance hit.
Does this mean that you could use Vulkan 1.3 to emulate the unified memory of many consoles without copying things from VRAM to RAM or vice versa without a performance penalty?

**Comment by Adamillo** [comment]
> > It helps now that modern APIs like Vulkan and DX12 have now finally closed that gap, with Vulkan having now direct RAM access with special API functions, so you can do things on CPU and GPU with zero performance hit.
> > Does this mean that you could use Vulkan 1.3 to emulate the unified memory of many consoles without copying things from VRAM to RAM or vice versa without a performance penalty?

**Comment by Jj0YzL5nvJ** [comment]
> > It helps now that modern APIs like Vulkan and DX12 have now finally closed that gap, with Vulkan having now direct RAM access with special API functions, so you can do things on CPU and GPU with zero performance hit.
> 
> Does this mean that you could use Vulkan 1.3 to emulate the unified memory of many consoles without copying things from VRAM to RAM or vice versa without a performance penalty?

From what I understand, those new APIs are smart enough to know when it is better to use RAM or VRAM. With previous APIs it was extremely easy to screw up and force the use of a context with the worst possible performance with certain GPUs.

See #1561 and #1960 as examples.

Note: @cruduxcru0 refers to [ParaLLEl RDP](https://github.com/Themaister/parallel-rdp#parallel-rdp) which states that it only needs Vulkan 1.1. The detail is that official support for the VK_KHR_8bit_storage extension is part of Vulkan 1.3.
Support for VK_KHR_8bit_storage in implementations prior to Vulkan 1.3 is unofficial. Current versions of ParaLLEl RDP avoid using the VK_KHR_8bit_storage / VK_KHR_16bit_storage extensions on _less efficient GPUs_, so the original claim is not false either... it can work with Vulkan 1.1, but Vulkan 1.3 is recommended.
In layman's terms, just buy NVIDIA.

**Comment by Adamillo** [comment]
I see, sounds awesome! Thanks for the insight on how things work. I'll close this now, since I'm pretty satisfied with the answers!
